java/algorithm/src/test/java/org/apache/arrow/algorithm/search/TestSortedVectorSearcher.java

```diff
@@ -0,0 +1,127 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.algorithm.search;
+
+import static org.junit.Assert.assertEquals;
+
+import org.apache.arrow.algorithm.sort.DefaultVectorComparators;
+import org.apache.arrow.algorithm.sort.FixedWidthOutOfPlaceVectorSorter;
+import org.apache.arrow.algorithm.sort.VariableWidthOutOfPlaceVectorSorter;
+import org.apache.arrow.algorithm.sort.VectorValueComparator;
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.memory.RootAllocator;
+import org.apache.arrow.vector.IntVector;
+import org.apache.arrow.vector.VarCharVector;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Test cases for {@link org.apache.arrow.algorithm.search.SortedVectorSearcher}.
+ */
+public class TestSortedVectorSearcher {
+
+  private final int VECTOR_LENGTH = 100;
+
+  private BufferAllocator allocator;
+
+  @Before
+  public void prepare() {
+    allocator = new RootAllocator(1024 * 1024);
+  }
+
+  @After
+  public void shutdown() {
+    allocator.close();
+  }
+
+  @Test
+  public void testSearchInt() {
+    try (IntVector rawVector = new IntVector("", allocator)) {
+      rawVector.allocateNew(VECTOR_LENGTH);
+      rawVector.setValueCount(VECTOR_LENGTH);
+
+      // prepare data in reverse order
+      for (int i = 0; i < VECTOR_LENGTH; i++) {
+        if (i == VECTOR_LENGTH - 1) {
+          rawVector.setNull(i);
+        } else {
+          rawVector.set(i, VECTOR_LENGTH - i - 1);
+        }
+      }
+
+      // sort vector
+      try (IntVector sortedVector = new IntVector("", allocator)) {
+        sortedVector.allocateNew(VECTOR_LENGTH);
+        sortedVector.setValueCount(VECTOR_LENGTH);
+
+        FixedWidthOutOfPlaceVectorSorter<IntVector> sorter = new FixedWidthOutOfPlaceVectorSorter<>();
+        VectorValueComparator<IntVector> comparator = new DefaultVectorComparators.IntComparator();
+        sorter.sortOutOfPlace(rawVector, sortedVector, comparator);
+
+        // do search
+        SortedVectorSearcher<IntVector> searcher = new SortedVectorSearcher<>(sortedVector, comparator);
+        for (int i = 0; i < VECTOR_LENGTH; i++) {
+          int result = searcher.search(rawVector, i);
+          assertEquals(VECTOR_LENGTH - 1 - i, result);
+        }
+      }
+    }
+  }
+
+  @Test
+  public void testSearchVarChar() {
+    try (VarCharVector rawVector = new VarCharVector("", allocator)) {
+      rawVector.allocateNew(VECTOR_LENGTH * 16, VECTOR_LENGTH);
+      rawVector.setValueCount(VECTOR_LENGTH);
+
+      byte[] content = new byte[2];
+
+      // prepare data in reverse order
```

Comment:
Why?  This seems to duplicate testing of the sorter?

Reply:
There is no particular reason for this.
I have revised the test case to prepare the data in sorted order. So there is no need to sort the data.